rest/diagnostic_doc_api.go

```diff
@@ -39,3 +56,64 @@ func (h *handler) handleGetDocChannels() error {
 	h.writeJSON(resp)
 	return nil
 }
+
+// HTTP handler for running a document through the sync function and returning the results
+// body only provided, the sync function will run with no oldDoc provided
+// body and doc ID provided, the sync function will run using the current revision in the bucket as oldDoc
+// docid only provided, if the doc exists in the bucket, sync fn results will be returned, otherwise it will error
+func (h *handler) handleSyncFnDryRun() error {
+	docid := h.getQuery("doc_id")
+
+	body, err := h.readDocument()
+	if err != nil {
+		if docid == "" {
+			return err
+		}
+	}
+	expiryPtr, channelSet, access, roles, err := h.collection.SyncFnDryrun(h.ctx(), body, docid)
+	errorMsg := ""
+	if err != nil {
+		errorMsg = err.Error()
+	}
+	expiry := uint32(0)
```

Comment:
We can't differentiate between Sync Functions that explicitly set expiry to zero and Sync Functions that don't set expiry with this handling.  I think Expiry needs to be *uint32 in SyncFnDryRun, potentially with omitempty?